Dockerfile

```
@@ -1,19 +1,34 @@
-
-FROM python:3.6
-
-WORKDIR /usr/src/app
-
-ENV APP_NAME respa
-
-RUN apt-get update && apt-get install -y gdal-bin postgresql-client
-
-COPY requirements.txt .
-COPY deploy/requirements.txt ./deploy/requirements.txt
-
-RUN pip install --no-cache-dir -r deploy/requirements.txt
-
-COPY . .
-
-RUN mkdir -p www/media
-
-CMD deploy/server.sh
+# ======================================================================
+FROM python:3.8-buster AS app-base
+# ======================================================================
+WORKDIR /app
+
+COPY --chown=appuser:appuser requirements.txt /app
+RUN groupadd --gid 1000 appuser \
+&& useradd --uid 1000 --gid appuser --create-home --shell /bin/bash appuser \
+&& apt-get update \
+&& apt-get install -y \
+    gdal-bin \
+    postgresql-client \
+&& curl -sL https://deb.nodesource.com/setup_12.x | bash - \
+&& apt-get install -y nodejs \
+&& pip install --no-cache-dir -r requirements.txt
```

Using `set -x` would provide better logging if things fail, and with `set -e` you could string commands together with `;` instead of `&&`:

```suggestion
RUN set -eux; \
    groupadd --gid 1000 appuser; \
    useradd --uid 1000 --gid appuser --create-home --shell /bin/bash appuser; \
    apt-get update; \
    apt-get install -y \
        gdal-bin \
        postgresql-client \
    ; \
    curl -sL https://deb.nodesource.com/setup_12.x | bash -; \
    apt-get install -y nodejs; \
    pip install --no-cache-dir -r requirements.txt; \
    ;
```

---

refactored